fix PartyMap MapRoom logic to not show room name when set to RoomVisibility.count

jigar-jstechno opened the pull request:
This fixes https://github.com/sparkletown/internal-sparkle-issues/issues/255

PartyMap MapRoom doesn't correctly handle `RoomVisibility` count/nameCount differences 
The bug is that we should then be rendering the content slightly differently based on these 2 values, yet currently we are treating them exactly the same:

- `RoomVisibility.count`: should only show the number of users in the room, and a 'person' icon
- `RoomVisibility.nameCount`: should show the room name AND the number of users in the room

0xdevalias commented:
@jigar-jstechno Can you please fix up the title of this issue to better represent what is being fixed, and add a quick summary description of the fix in the PR description. Linking to the internal issue like you did with the closes keywords is good (though I changed it to 'fixes' rather than 'closes' since this is for a bug fix), but since the internal issues are private, and this repo is public, having a brief summary of the issue here would be good so that the open-source community is able to see what the fix relates to. It would also be good to add the `bug` label here too.

0xdevalias reviewed (changes requested):
@jigar-jstechno It looks like there's a whole bunch of unrelated code in this PR. I'm seeing changes from the notification sound PR, the footer fix, etc.

Did you start this PR's branch from an updated/clean clone of the `staging` branch?

jigar-jstechno commented:
> @jigar-jstechno It looks like there's a whole bunch of unrelated code in this PR. I'm seeing changes from the notification sound PR, the footer fix, etc.
> 
> Did you start this PR's branch from an updated/clean clone of the `staging` branch?

i think it is merged with previous one. should I close this pr and create new ?

jigar-jstechno commented:
> @jigar-jstechno Can you please fix up the title of this issue to better represent what is being fixed, and add a quick summary description of the fix in the PR description. Linking to the internal issue like you did with the closes keywords is good (though I changed it to 'fixes' rather than 'closes' since this is for a bug fix), but since the internal issues are private, and this repo is public, having a brief summary of the issue here would be good so that the open-source community is able to see what the fix relates to. It would also be good to add the `bug` label here too.

Sure will add .

jigar-jstechno reviewed (commented):
Hey @mike-lvov @0xdevalias  
Check now as i removed unwanted files.

0xdevalias reviewed (changes requested):
It looks like you may have entirely deleted the files, rather than just resetting them to the state they were in before.

You can try doing something like 'git checkout staging -- path/to/file' or similar methods to reset these files back to their 'unmodified state'

0xdevalias commented on src/components/templates/PartyMap/components/Map/MapRoom.scss:
```diff
@@ -29,6 +29,18 @@ $label-visible: flex;
     }
   }
 
+  &.maproom--always-show-label-count {
+    .maproom__label {
+      display: $label-visible;
+    }
+  }
+
+  &.maproom--always-show-label-count {
+    .maproom__roomtitle {
+      display: $label-hidden;
+    }
+  }
```
While I can see the approach you were going for here, I think there's a simpler way we can implement this feature (see other comments in this review)

0xdevalias replied:
FYI, this pattern of CSS is called BEM (Block, Element, Modifier), which you can read more about at:

- http://getbem.com/naming/

In a majority of cases, we would stick to setting the 'modifier' (eg. `--always-show-label`) on the 'block' (eg. `.maproom`), as it seems you have followed here. Sometimes (if needed), we can also set it on the element itself (eg. `.maproom__title--foo`).

0xdevalias commented on src/components/templates/PartyMap/components/Map/MapRoom.tsx:
```diff
@@ -60,7 +61,7 @@ export const MapRoom: React.FC<MapRoomProps> = ({
 
       <div className="maproom__label">
-        {room.title}
+        <span className="maproom__roomtitle">{room.title}</span>
```
I think we can be even simpler here, and just conditionally render the `room.title` in the React/JSX directly. No need to add the extra CSS/etc for it.

You can read more about this here:

- https://reactjs.org/docs/conditional-rendering.html

Specifically this part:

- https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

---

By doing it this way, you can then probably clean up/remove all of the extra CSS changes that you made in this PR.

0xdevalias reviewed (changes requested):
Further to my last review, some comments on the implementation itself.

0xdevalias reviewed (changes requested):
Thanks for making the changes requested in https://github.com/sparkletown/sparkle/pull/1186#pullrequestreview-592731955 , this part of the code looks good now, and this is the solution I had in mind.

Though it looks like the changes requested in https://github.com/sparkletown/sparkle/pull/1186#pullrequestreview-592587292 are still outstanding, which is why the diff here is showing 4 deleted files, rather than the single file you made your code change to. These files will need to be restored before this can be merged, as described in my previous review comments.

0xdevalias commented:
RE: https://github.com/sparkletown/sparkle/pull/1186#issuecomment-780363785

> > Can you please fix up the title of this issue to better represent what is being fixed, and add a quick summary description of the fix in the PR description. Linking to the internal issue like you did with the closes keywords is good (though I changed it to 'fixes' rather than 'closes' since this is for a bug fix), but since the internal issues are private, and this repo is public, having a brief summary of the issue here would be good so that the open-source community is able to see what the fix relates to. It would also be good to add the bug label here too.
> 
> Sure will add

@jigar-jstechno It looks like you updated the description, thanks for that. Though it looks like you haven't fixed up the title or labels as requested yet. Can you please make sure to do that.

jigar-jstechno commented:
> It looks like you may have entirely deleted the files, rather than just resetting them to the state they were in before.
> 
> You can try doing something like 'git checkout staging -- path/to/file' or similar methods to reset these files back to their 'unmodified state'

i run this command git checkout staging -- path/to/file'  but my staging branch have those new changes in PR #1180 done so if i commit all file 6 file will display here... any solution for this?

jigar-jstechno commented:
> RE: [#1186 (comment)](https://github.com/sparkletown/sparkle/pull/1186#issuecomment-780363785)
> 
> > > Can you please fix up the title of this issue to better represent what is being fixed, and add a quick summary description of the fix in the PR description. Linking to the internal issue like you did with the closes keywords is good (though I changed it to 'fixes' rather than 'closes' since this is for a bug fix), but since the internal issues are private, and this repo is public, having a brief summary of the issue here would be good so that the open-source community is able to see what the fix relates to. It would also be good to add the bug label here too.
> > 
> > 
> > Sure will add
> 
> @jigar-jstechno It looks like you updated the description, thanks for that. Though it looks like you haven't fixed up the title or labels as requested yet. Can you please make sure to do that.

updated title

jigar-jstechno commented:
> > It looks like you may have entirely deleted the files, rather than just resetting them to the state they were in before.
> > You can try doing something like 'git checkout staging -- path/to/file' or similar methods to reset these files back to their 'unmodified state'
> 
> i run this command git checkout staging -- path/to/file' but my staging branch have those new changes in PR #1180 done so if i commit all file 6 file will display here... any solution for this?

Solved this now It shows only 1 file changed

0xdevalias commented:
> > It looks like you may have entirely deleted the files, rather than just resetting them to the state they were in before.
> > You can try doing something like 'git checkout staging -- path/to/file' or similar methods to reset these files back to their 'unmodified state'
> 
> i run this command git checkout staging -- path/to/file' but my staging branch have those new changes in PR #1180 done so if i commit all file 6 file will display here... any solution for this?

@jigar-jstechno I believe that's because one of your previous PRs was made directly from `staging`, rather than making a new branch for it.

I think you would want to remove your local `staging` branch, then check it out from our repo again. If you haven't already, you may need to add this repo as a remote, perhaps called `upstream`, and then check the `staging` branch out from there.

- https://docs.github.com/en/github/using-git/adding-a-remote
  - `git remote add upstream git@github.com:sparkletown/sparkle.git`
- https://stackoverflow.com/questions/1783405/how-do-i-check-out-a-remote-git-branch/1783426#1783426
  - `git checkout -b staging upstream/staging`
  
If you were just wanting to grab these few files from `upstream/staging`, then I think you could probably do something like:

- `git checkout upstream/staging -- path/to/file.ext`

0xdevalias commented:
> > It looks like you updated the description, thanks for that. Though it looks like you haven't fixed up the title or labels as requested yet. Can you please make sure to do that.
> 
> updated title

@jigar-jstechno Thanks for adding the title. Now there is just the labels left to do, as per my original request/last followup request:

> It would also be good to add the bug label here too.

jigar-jstechno commented:
> @jigar-jstechno Can you please fix up the title of this issue to better represent what is being fixed, and add a quick summary description of the fix in the PR description. Linking to the internal issue like you did with the closes keywords is good (though I changed it to 'fixes' rather than 'closes' since this is for a bug fix), but since the internal issues are private, and this repo is public, having a brief summary of the issue here would be good so that the open-source community is able to see what the fix relates to. It would also be good to add the `bug` label here too.

Yeah I added bug label also i want  to ask if there any documentation for label which label is used in which scenario so in future PR I will take care of this.

0xdevalias commented:
> Yeah I added bug label also i want to ask if there any documentation for label which label is used in which scenario so in future PR I will take care of this.

@jigar-jstechno The closest we have to documentation on this at the moment is https://github.com/sparkletown/internal-sparkle-issues/issues/337, specifically https://github.com/sparkletown/internal-sparkle-issues/issues/337#issuecomment-779170484 for PRs

0xdevalias reviewed (approved):
Your last changes in https://github.com/sparkletown/sparkle/pull/1186/commits/1dcf70233cce3e1310859c476cf5e416a3f60548 were basically what I was thinking/wanting to see for this feature.

Though as I was about to approve it, I decided to do a quick clickthrough to check it all worked correctly, and noticed that when there are no users in the room, the label displays (but with no content in it), which we don't want:

![image](https://user-images.githubusercontent.com/753891/108317991-38556e00-7213-11eb-9e8d-c3bcecf3d8ec.png)

I had to play around a little bit to get a good solution for it, so I ended up pushing this directly to your branch in https://github.com/sparkletown/sparkle/pull/1186/commits/4c5bc25f782d366f48ce94285cd2f211c6797a8b

---

I then noticed that the label was really wide when there was only a single user, and the text wasn't centred; which looked weird:
![image](https://user-images.githubusercontent.com/753891/108318372-d5b0a200-7213-11eb-9d4a-446000a0ac2f.png)

![image](https://user-images.githubusercontent.com/753891/108318319-bd408780-7213-11eb-8951-22b068f338e0.png)

So I fixed that in https://github.com/sparkletown/sparkle/pull/1186/commits/da50a75bf744942b2a1faa2fa24435ae77ab58a5, so it now looks like this when not hovered:

![image](https://user-images.githubusercontent.com/753891/108318449-f7118e00-7213-11eb-9865-3357bd283181.png)

And this when hovered:

![image](https://user-images.githubusercontent.com/753891/108318502-0bee2180-7214-11eb-8e09-767ed5da2937.png)

---

I figured it would be quicker to push these changes directly.

With those changes added, I think this PR is good to go now!

0xdevalias commented:
@jigar-jstechno Now that this is approved (see https://github.com/sparkletown/sparkle/pull/1186#pullrequestreview-592905054), you can click on the 'squash and merge' button to merge it into `staging`, which will also close this PR. Then make sure you go back to https://github.com/sparkletown/internal-sparkle-issues/issues/255 and make sure you updat the 'project' entry to 'Ready for QA'
